vnext/Shared/MemoryMappedBuffer.cpp

```diff
@@ -0,0 +1,154 @@
+#include "pch.h"
+#include "MemoryMappedBuffer.h"
+
+#include <werapi.h>
+#include <windows.h>
+
+namespace {
+
+class MemoryMappedBuffer : public facebook::jsi::Buffer {
+ public:
+  MemoryMappedBuffer(const wchar_t *const filename, uint32_t offset);
+
+  size_t size() const override;
+  const uint8_t *data() const override;
+
+ private:
+  static void fileDataDeleter(void *p) {
+    WerUnregisterMemoryBlock(p);
+    UnmapViewOfFile(p);
+  }
+
+  std::unique_ptr<void, decltype(&CloseHandle)> m_fileMapping;
+  std::unique_ptr<void, decltype(&fileDataDeleter)> m_fileData;
+  uint32_t m_size;
+  uint32_t m_offset;
+};
+
+MemoryMappedBuffer::MemoryMappedBuffer(
+    const wchar_t *const filename,
+    uint32_t offset)
+    : m_fileMapping{nullptr, &CloseHandle},
+      m_fileData{nullptr, &fileDataDeleter},
+      m_size{0},
+      m_offset{0} {
+#if (defined(WINRT))
+  std::unique_ptr<void, decltype(&CloseHandle)> fileHandle{
+      CreateFile2(
```

Comment:
>CreateFile2 [](start = 6, length = 11)

Please add comment why we are using different Apis on Win32 and WinRt. Ideally, we would use the same Api

Reply:
Added some comments; please review.

---
In reply to: [320882173](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320882173) [](ancestors = 320882173)